### PR TITLE
[WIP]Vehicle deconstruct zone for NPCs to work on

### PR DESCRIPTION
**Pull request by ghost**
#### Summary
```SUMMARY: Features "Vehicle deconstruct zone for NPCs to work on"```

#### Purpose of change
Allows follower NPCs to deconstruct a vehicle for you in real-time.
THIS REQUIRES #33091 AND HAS THE SAME CHANGES FROM IT

#### Describe the solution
Adds a Vehicle Deconstruct Zone to the zone manager, and a talk function to NPCs to order them to start working on deconstructing the vehicle. 
They will go and fetch a tool that is required, and get to work, then tidy up all the parts to a nearby unsorted zone,

#### Describe alternatives you've considered
N/A

#### Additional context
This will eventually replace the faction camp chop shop mission.
Gif of it in action - https://gfycat.com/clutteredinformaldolphin

### CODE REVIEW

**Comment by davidpwbrown** [comment]
Ok I didnt realise #33673 haad been merged, ill need to adapt to that.

File: src/vehicle.cpp [hunk 1]
```diff
@@ -2536,6 +2536,30 @@ std::vector<int> vehicle::all_parts_at_location( const std::string &location ) c
...
+// as the part index was first "chosen" before the NPC started travelling here.
+// therefore the part index is now invalid shifted by one or two ( depending on how many other NPCs working on this vehicle )
+// so loop over the part indexes in reverse order to get the next one down that matches the part type we wanted to remove
+int vehicle::get_next_shifted_index( int original_index, player &p )
+{
+    int ret_index = original_index;
+    bool found_shifted_index = false;
+    for(std::vector<vehicle_part>::reverse_iterator it = parts.rbegin(); it != parts.rend(); ++it) {
+        if( p.get_value( "veh_index_type" ) == it->info().name() ){
+            ret_index = index_of_part( &(*it) );
```
Comment (BevapDin) [inline src/vehicle.cpp]:
```suggestion
            ret_index = index_of_part( &*it );
```

File: src/player.h [hunk 2]
```diff
@@ -1515,6 +1515,7 @@ class player : public Character
         // Auto move methods
         void set_destination( const std::vector<tripoint> &route,
                               const player_activity &destination_activity = player_activity() );
+        void destination_activity_value( const int value );
```
Comment (BevapDin) [inline src/player.h]: Where is this function implemented?

Comment (davidpwbrown) [reply src/player.h]: Nowhere, thats a leftover, ill remove it.

**Comment by davidpwbrown** [comment]
sorry - closing this, and redoing a new PR, as it was easier to reconstruct the branch instead of hassling through the merge conflicts from rebasing after #33673

**Pull request closed without merging**